docs/webui_overriding.md

```diff
@@ -0,0 +1,139 @@
+# Overriding Upstream WebUIs
+
+Currently, there are a number of ways to override upstream WebUIs. Which one
+you want will depend on what you're trying to do, and how the upstream WebUI is
+implemented.
+
+## chromium_src overrides
+
+This is probably the simplest approach. Similarly to our C++ overrides just add
+a new file at the same path but in the `chromium_src` director and it will
+replace the upstream version of the file. The upstream version can be imported
+via `./<filename>-chromium.ext` if you need to reexport something from it.
+
+**Note:** You don't need to reexport upstream stuff if you're completely
+replacing it.
+
+### Example Typescript
+
+```ts
+// in chromium_src/path/to/file.ts
+import { frob as frobChromium } from './file-chromium.js'
+
+// export everything from upstream
+export * from './file-chroimum.js'
+
+// in Brave, all frobs are twice as good.
+export function frob() {
+  return frobChromium() * 2
+}
+```
+
+### Example CSS
+
+Note: This is slightly different depending on whether you're overriding a Lit
+CSS file or a Polymer CSS file (you need to make sure `type` is correct in the
+`css_wrapper_metadata`)
+
+```css
+/* in chromium_src/path/to/file.css */
+
+/* #css_wrapper_metadata_start
+ * #type=style-lit
+ * #import=./file-chromium.css.js
+ * #scheme=relative
+ * #css_wrapper_metadata_end */
+
+:host {
+  background: hotpink;
+}
+```
+
+## Lit Mangling
+
+Unfortunately, the above strategies don't work for modifying Lit HTML. To modify
+a Lit HTML template we need a way to directly modify the source file. This is
+done with a Lit mangler.
+
```

Comment:
Well, chromium_src overriding could "work" but isn't as fine-tunable. Either way, maybe we should simply start with the second sentence?

Reply:
They above will work to completely replace the HTML file - they won't work at all for modifying it I don't think.